test/doctrines/Preorders.jl

```diff
@@ -0,0 +1,52 @@
+using Test
+using Catlab
+using Catlab.Doctrines
+
+x,y,z = Ob(FreeThinCategory, :x, :y, :z)
+f, g = Hom(:f, x, y), Hom(:g, y, z)
+@test dom(compose(f,g)) == x
+@test codom(compose(f,g)) == z
+
+
+x,y,z = Ob(FreePreorder, :x, :y, :z)
+@test_throws UndefVarError f, g = Leq(:f, x, y), Leq(:g, y, z)
+# Preorder doesn't have domain and codomain defined.
+@test_throws UndefVarError dom(transitive(f,g)) == x
```

Comment:
Because `Leq` is defined as `Leq(lhs::Elt, rhs::Elt)`, the accessors are `lhs` and `rhs`, not `dom` and `codom`. So `lhs(transitive(f,g)) == x` and `rhs(transitive(f,g)) == z` should work.

That reminds me: we should also export `lhs` and `rhs`.